src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,95 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+const initialMovie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [newMovie, setNewMovie] = useState<Movie>(initialMovie);
+  const [count, setCount] = useState(0);
+
+  const isReadyToSubmit =
+    newMovie.title.trim() &&
+    newMovie.imgUrl.trim() &&
+    newMovie.imdbId.trim() &&
+    newMovie.imdbUrl.trim();
+
+  const handleChange = (key: string, value: string) => {
+    setNewMovie({
+      ...newMovie,
+      [key]: value,
+    });
+  };
```

Comment:
we can even simplify this handler using `event` where we can get `name` and `value`
```suggestion
  const handleChange = (e) => {
    const {name, value} = e.target;
    
    setNewMovie({
      ...newMovie,
      [name]: value,
    });
  };
```